website/database.py

```diff
@@ -608,6 +611,52 @@ def batch_get_adventures(self, adventure_ids):
         keys = {id: {"id": id} for id in adventure_ids}
         return ADVENTURES.batch_get(keys) if keys else {}
 
+    def get_public_adventures_filters(self):
+        filters = {}
+        for record in PUBLIC_ADVENTURES_FILTERS.scan():
+            if record.get("field") == "tag":
+                filters["tag"] = filters.get("tag", [])
+                filters["tag"].append(record.get("value"))
+            else:
+                lang, level = record.get("value").split("#")
+                filters["lang"] = filters.get("lang", [])
+                filters["lang"].append(lang)
+                filters["level"] = filters.get("level", [])
+                filters["level"].append(level)
+        return filters
+
+    def update_public_adventure_filters_indexes(self, adventure):
```

Comment:
True, but i use it in other places; e.g., when deleting a tag or cloning an adventure!

Reply:
What I'm getting at is that you should hide the details of updating indexes as much as possible.

If the front-end doesn't need to know about the fact that there are indexes, if this can all remain in the `Database` class -- so much the better! Less for consumers to keep in mind, fewer chances of bugs.

Remember that there are other people working on this code base, who might not know the intricacies of "when you call function A, don't forget to also call function B". Ideally, they shouldn't have to! And don't underestimate how much you yourself will have forgotten about the details in 3 months' time as well 😉.

So, any operation that needs to touch the indexes should go through a public method of the `Database` class anyway; and that method can call this one whenever it needs to.

Reply:
Still relevant.